template/web/src/hooks/useBlockNumber.ts

```diff
@@ -1,56 +1,41 @@
-import { useState, useEffect } from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { Chain } from 'viem/chains';
 import { useAccount } from 'wagmi';
 
 type BlockNumberResponse = {
   block: number;
 };
 
-const useCurrentBlockNumber = (refreshIntervalMs = 10000) => {
-  const [blockNumber, setBlockNumber] = useState(0);
-  const [isLoading, setIsLoading] = useState(true);
-  const { chain } = useAccount();
-  const chainId = chain?.id;
+type Props = {
+  refetchInterval?: number;
+  enabled?: boolean;
+  chainId?: Chain['id'];
+};
 
-  useEffect(() => {
-    let intervalId: NodeJS.Timeout;
+//TODO: consider using wagmi's `useBlockNumber` instead
+const useCurrentBlockNumber = (props?: Props) => {
```

Comment:
Whats the reasoning behind getting the chainId from the arguments if present, why not always take the one connected to the wallet?

Reply:
Maybe it's a personal preference, but I think it's a bad UX where user needs to switch chain to view your dapp. This is more important for write operation of course, but reads as well. For example you have an NFT dapp that supports NFTs from multiple chains, why would user need to switch chains every time he opens an NFT details page of a different chain from the one connected in a wallet, just for a page to read some contract states.

Perhaps for this particular hook it is not as needed, but I think it's a good practice and ideally all wagmi hook wrappers should always have an option to pass chainId explicitly

Reply:
Makes sense. I think this might cause confusion to people who are reading this for the first time on why the chainId is added as a prop for this particular hook and not for all the other read hooks. Just so that we maintain a pattern, we should maybe not add this right now and later we can do a more bigger cleanup where we can add chainId to all read/write hooks. Thoughts? 

CC @Zizzamia @kyhyco @cnasc feel free to add your thoughts as well

Reply:
ok makes sense. I can create a discussion or an issue instead

Reply:
btw, I added a comment here also as I think this hook is completly redundant and it's better to just use wagmi's hook

Reply:
I would agree that in practice people should use the `wagmi` hook. If I remember correctly, this hook was intended only as a beginner example to show some code using `wagmi` in action. This library has evolved quite a bit since when this was written and I don't think it serves the purpose it once did. IMO would +1 on removing it, interested in hearing from the others

Reply:
3, 2, 1, and it's gone
https://github.com/coinbase/build-onchain-apps/pull/460